realm/realm-library/src/main/cpp/java_binding_context.cpp

```diff
@@ -17,18 +17,40 @@
 #include "java_binding_context.hpp"
 #include "jni_util/java_method.hpp"
 
+#include "util.hpp"
+
 using namespace realm;
 using namespace realm::_impl;
 using namespace realm::jni_util;
 
-void JavaBindingContext::changes_available()
+void JavaBindingContext::before_notify()
 {
+    if (JniUtils::get_env()->ExceptionCheck()) {
+        return;
```

Comment:
We cannot, and that is the annoy part. `before_notify` and `did_change` will be called in the same ALooper event loop, and we don't have chance to return to jvm until the ALooper callback returns.